### DISCUSSION TITLE
Disable Trimming Text to View Height

### DISCUSSION THREAD

**Discussion by c1moore** [charmbracelet/bubbletea#1179] [Q&A]
I'm trying to create a tool that has a dry run or plan mode.  This dry run mode can print out quite a few lines depending on the number of changes it will perform (think similar to Terraform output).

I'd like the end result to look something like this:

```sh
 ✓ Loaded state

# ...plan details...

150 things to do
```

I've found 2 ways to go about printing out this plan, both seem flawed:
1. Use `Program.Printf()`
2. Create a "static text" model

The problem with 1 is any components that were rendered before I use `Printf` as actual Models are moved below the text.  So for example, you may see something like

```sh
# ...plan details...

150 things to do
 ✓ Loaded state
```

The above example doesn't look terrible, but it's a little more confusing for my usecase.

For 2, it appears that bubbletea trims the output to the height of the terminal.  So if the terminal can only display 10 lines, you'll only ever see the last 10 lines of the plan.

So what would be the recommended way to handle the above?

**Comment by NPellet** [comment]
We have the exact same issue.
Initially we started using ViewPort for it, but the problem is that we need to enable scrolling, which disables text selection.
So now we'd like to move away from ViewPort again, but the trimming prohibits us from doing so
